### PR TITLE
for review: proposed solution for #2620 and #2778

### PR DESCRIPTION
**Pull request by griebd**
after #2763 and #2766, here it's again... :wink:
I will focus on the two issues at hand, the part of the cout's I'll do later, after finishing this one.

It still needs a lot of changes on the existing code but the new part is done! For a taranis eeprom it's working already.

What do you think of this solution?

### CODE REVIEW

File: companion/src/eeprominterface.cpp [hunk 1]
```diff
@@ -1357,21 +1357,74 @@ QList<FirmwareInterface *> firmwares;
...
-void UnregisterFirmwares() 
+void UnregisterFirmwares()
 {
   foreach (FirmwareInterface * f, firmwares) {
     delete f;
   }
 }
 
-bool LoadEeprom(RadioData &radioData, const uint8_t *eeprom, const int size)
+void ShowEepromErrors(QWidget *parent, QString title, QString mainMessage, unsigned long errorsFound)
```
Comment (bsongis) [inline companion/src/eeprominterface.cpp]: const QString & here!

File: companion/src/eeprominterface.cpp [hunk 2]
```diff
@@ -1357,21 +1357,74 @@ QList<FirmwareInterface *> firmwares;
...
+{
+  std::bitset<NUM_ERRORS> errors(errorsFound);
+  QString errorsList(QT_TRANSLATE_NOOP("EepromInterface", "Possible causes for this:\n"));
+
+  if (errors.test(UNSUPPORTED_NEWER_VERSION)) { errorsList += QT_TRANSLATE_NOOP("EepromInterface", "\n- Eeprom is from a newer version of OpenTX"); }
```
Comment (bsongis) [inline companion/src/eeprominterface.cpp]: Instead of the \n in all strings, I would see a QStringList with join("\n"), no?

**Comment by bsongis** [comment]
Why not, I will merge the pull request and you will fix the little things after, ok? It will avoid merge problems ...

**Comment by griebd** [comment]
ok, working to finish it then...

**Comment by bsongis** [comment]
Is it possible to release in this state? Or do you prefer we wait some more days?

**Comment by griebd** [comment]
I think there will be problems as it is... I didn't finish it yet because a friend of my got shot at work a few days ago, he is recovering now and I'm coming back to normal life! I should be able to finish it this week, I think, so, wait a few days, please.

**Comment by kilrah** [comment]
Ouch!

**Pull request merged**